test/integration/testpaths/testpath_volume_recurring_snap.py

```diff
@@ -0,0 +1,1015 @@
+            # Deploy A VM
+            cls.vm_1 = VirtualMachine.create(
+                cls.userapiclient,
+                cls.testdata["small"],
+                templateid=cls.template.id,
+                accountid=cls.account.name,
+                domainid=cls.account.domainid,
+                serviceofferingid=cls.service_offering.id,
+                zoneid=cls.zone.id,
+                diskofferingid=cls.disk_offering.id,
+                mode=cls.zone.networktype
+            )
+
+            cls.volume = list_volumes(
+                cls.apiclient,
+                virtualmachineid=cls.vm_1.id,
+                type='ROOT',
+                listall=True
+            )
+
+            cls.data_volume = list_volumes(
+                cls.apiclient,
+                virtualmachineid=cls.vm_1.id,
+                type='DATADISK',
+                listall=True
+            )
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"])
+    def test_01_volume_snapshot(self):
+        """ Test Volume (root) Snapshot
+        # 1. Create Hourly, Daily,Weekly recurring snapshot policy for ROOT disk and \
+                    Verify the presence of the corresponding snapshots on the Secondary Storage
+        # 2. Delete the snapshot policy and verify the entry as Destroyed in snapshot_schedule
+        # 3. Verify that maxsnaps should not consider manual snapshots for deletion
+        # 4. Snapshot policy should reflect the correct timezone
+        # 5. Verify that listSnapshotPolicies() should return all snapshot policies \
+                that belong to the account (both manual and recurring snapshots)
+        # 6. Verify that listSnapshotPolicies() should not return snapshot \
+                policies that have been deleted
+        # 7. Verify that snapshot should not be created for VM in Destroyed state
+        # 8. Verify that snapshot should get created after resuming the VM
+        # 9. Verify that All the recurring policies associated with the VM should be \
+                deleted after VM get destroyed.
+        """
+        # Step 1
+        recurring_snapshot = SnapshotPolicy.create(
+            self.apiclient,
+            self.volume[0].id,
+            self.testdata["recurring_snapshot"]
```

Comment:
What is the snapshot policy interval here? Rather than passing same data as in test data, define the data in the test case itself whichever interval is want, and pass that data to snapshot policy. The test case behavior should not change if the dict in test_data.py is changed in future.